Bottom: 66702d5a96597802b71c185c76857f89d0ac0c82
Top:    4afe09502daedbc902029e91a3483862b2e4d24b
Author: Ian Kent <raven@themaw.net>
Date:   2013-07-30 14:09:56 +0800

Refresh of autofs-5.0.7-add-dumpmap-to-file.patch

---

diff --git a/daemon/automount.c b/daemon/automount.c
index 9704ceb..bbde805 100644
--- a/daemon/automount.c
+++ b/daemon/automount.c
@@ -1948,7 +1948,7 @@ int main(int argc, char *argv[])
 		{"foreground", 0, 0, 'f'},
 		{"random-multimount-selection", 0, 0, 'r'},
 		{"negative-timeout", 1, 0, 'n'},
-		{"dumpmaps", 1, 0, 'm'},
+		{"dumpmaps", 0, 0, 'm'},
 		{"global-options", 1, 0, 'O'},
 		{"version", 0, 0, 'V'},
 		{"set-log-priority", 1, 0, 'l'},
diff --git a/lib/master.c b/lib/master.c
index 22a289e..7f4d52e 100644
--- a/lib/master.c
+++ b/lib/master.c
@@ -1287,7 +1287,7 @@ static char *match_map_name(const char *match, const char *maps)
 	char *map_name;
 	char *this;
 	char *tok, *ptr, *tmp;
-	unsigned int found;
+	unsigned int found = 0;
 
 	tmp = strdup(match);
 	if (tmp)
@@ -1295,6 +1295,8 @@ static char *match_map_name(const char *match, const char *maps)
 	if (!tmp || !names) {
 		if (tmp)
 			free(tmp);
+		/* Don't print "not found" message since we haven't looked */
+		found = 1;
 		goto fail;
 	}
 
@@ -1303,7 +1305,6 @@ static char *match_map_name(const char *match, const char *maps)
 	else
 		map_name = tmp;
 
-	found = 0;
 	this = NULL;
 	ptr = NULL;
 	tok = strtok_r(names, ",", &ptr);
